### DISCUSSION TITLE
Regression since extension clients - SelectSubset type to narrow the exact selection is not possible anymore

### DISCUSSION THREAD

**Discussion by IzioDev** [prisma/prisma#20555] [Q&A]
### Question

How to benefit from `SelectSubset` utility type to correctly infer `return selection` with an extended client without having type error?

[Full Slack Issue Description](https://prisma.slack.com/archives/CA491RJH0/p1691070369939209)

### How to reproduce (optional)

Create a wrapper function that abstract extended `prisma` calls:
```ts
const decoratedPrisma = getExtendedPrismaClient(new PrismaClient())

function getDiscoveryScore<T extends Prisma.DiscoveryScoreFindUniqueArgs>(
    args: Prisma.SelectSubset<T, Prisma.DiscoveryScoreFindUniqueArgs>
) {
    return decoratedPrisma.discoveryScore.findUnique<
        Prisma.SelectSubset<T, Prisma.DiscoveryScoreFindUniqueArgs>
    >(args);
}
```

Note: this works before upgrading to `5.x` from `4.10.1`.

* See the diagnosis type error:
![image (18)](https://github.com/prisma/prisma/assets/9900846/520d2ea5-bf2c-42ba-8634-c0c186f1c73a)

### Expected behavior (optional)

I expect a utility type, like `SelectSubset` but on the `ExtendedClient` type, not on the `default` `PrismaClient` (I understand signature may have changed during the extensions registering).

### Information about Prisma Schema, Client Queries and Environment (optional)

Prisma Version: `5.1.0`

other informations are not needed, I think, let me know.

**Comment by nikolasburk** [comment]
**Just for context, I'm adding the full issue description from [Slack](https://prisma.slack.com/archives/CA491RJH0/p1691070369939209) here as well:**

Hi,
Since we've upgraded to 5.x we're having troubles to mimic the Extended Client Types in low-level repositories.
What we had (simplified for the sake of the example):

```ts
function getDiscoveryScore<T extends Prisma.DiscoveryScoreFindUniqueArgs>(
    args: Prisma.SelectSubset<T, Prisma.DiscoveryScoreFindUniqueArgs>
) {
    return prisma.discoveryScore.findUnique<
        Prisma.SelectSubset<T, Prisma.DiscoveryScoreFindUniqueArgs>
    >(args);
}
```

this way, the caller (functions that call this one) have the exact `select` type they requested as return, thanks to `SelectSubset`
Now it is not possible to do the same thing when using a decorated instance, example decoration:

```ts
export const decoratePrismaMethods = (datasourceAccessor: PrismaClient) => {
    return datasourceAccessor
        .$extends(extendsMessageMethods(datasourceAccessor))
        .$extends(extendsChunkMethods(datasourceAccessor));
};
export type DecoratedPrismaClient = ReturnType<typeof decoratePrismaMethods>;
```

And now, not working code using this decorated instance:

```ts
const decoratedPrisma = decoratePrismaMethods(new PrismaClient())

function getDiscoveryScore<T extends Prisma.DiscoveryScoreFindUniqueArgs>(
    args: Prisma.SelectSubset<T, Prisma.DiscoveryScoreFindUniqueArgs>
) {
    return decoratedPrisma.discoveryScore.findUnique<
        Prisma.SelectSubset<T, Prisma.DiscoveryScoreFindUniqueArgs>
    >(args);
}
```

LSP Error Diagnosis in attached image (it's Neovim btw :sunglasses:)
Is there something to do in particular, we've tried to take the `DiscoveryScoreFindUniqueArgs` from the `DecoratedClient` but it seems not possible to do so. We can get the function signature: `DecoratedClient["Entity"]["Function"]` but not the args from them.

**Reply by ludralph** [reply]
Hi @IzioDev 👋 

I can confirm this regression as I tested with Prisma version 4.10.1 and it worked but throws a compile time error with version 5.1.1. Reproduction can be seen [here](https://github.com/ludralph/SelectSubset-reproduction)

Can you please open a bug report using this [link](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by ABitShift** [reply]
Dude, there is literally a reproduction repository created by one of your member, how could you say it's outdated?
